File: algorithms/1020. Number of Enclaves/solution.cc
class Solution {
public:
    void dfs(vector<vector<int>>& g, int i, int j) {
        if (0 <= i && i < g.size() && 0 <= j && j < g[i].size() && g[i][j]) {
            g[i][j] = 0;
            dfs(g, i-1, j);
            dfs(g, i, j+1);
            dfs(g, i+1, j);
            dfs(g, i, j-1);
        }
    }
    int numEnclaves(vector<vector<int>>& A) {
        vector<vector<int>> g = A;
        for (int i = 0; i < g.size(); ++i) {
            for (int j = 0; j < g[0].size(); ++j) {
                if (i == 0 || j == 0 || i == g.size() - 1 || j == g[0].size() - 1)
                    dfs(g, i, j);
            }
        }
        int result = 0;
        for (int i = 0; i < g.size(); ++i) {
            for (int j = 0; j < g[0].size(); ++j) {
                result += g[i][j];
            }
        }
        return result;
    }
};

File: algorithms/1020. Number of Enclaves/solution.py
class Solution:
    def numEnclaves(self, A: List[List[int]]) -> int:
        m, n = len(A), len(A[0])
        g = copy.deepcopy(A)
        def dfs(i, j):
            g[i][j] = 0
            for x, y in ((i-1, j), (i, j+1), (i+1, j), (i, j-1)):
                if 0 <= x < m and 0 <= y < n and g[x][y] == 1:
                    dfs(x, y)
        for i in range(m):
            for j in range(n):
                if (i == 0 or j == 0 or i == m - 1 or j == n - 1) and g[i][j] == 1:
                    dfs(i, j)
        return sum(sum(row) for row in g)
